internal/proto/local/controller/api/resources/credentialstores/v1/credential_store.proto

```diff
@@ -0,0 +1,69 @@
+syntax = "proto3";
+
+package controller.api.resources.credentialstores.v1;
+
+option go_package = "github.com/hashicorp/boundary/internal/gen/controller/api/resources/credentialstores;credentialstores";
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/struct.proto";
+import "controller/api/resources/scopes/v1/scope.proto";
+import "controller/custom_options/v1/options.proto";
+
+// CredentialStore contains all fields related to an Credential Store resource
+message CredentialStore {
+  // Output only. The ID of the Credential Store.
+  string id = 10;
+
+  // The ID of the Scope of which this Credential Store is a part.
+  string scope_id = 20 [json_name = "scope_id"];
+
+  // Output only. Scope information for this Credential Store.
+  resources.scopes.v1.ScopeInfo scope = 30;
+
+  // Optional name for identification purposes.
+  google.protobuf.StringValue name = 40 [(custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = { this: "name" that: "Name" }];
+
+  // Optional user-set description for identification purposes.
+  google.protobuf.StringValue description = 50 [(custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = { this: "description" that: "Description" }];
+
+  // Output only. The time this resource was created.
+  google.protobuf.Timestamp created_time = 60 [json_name = "created_time"];
+
+  // Output only. The time this resource was last updated.
+  google.protobuf.Timestamp updated_time = 70 [json_name = "updated_time"];
+
+  // Version is used in mutation requests, after the initial creation, to ensure this resource has not changed.
+  // The mutation will fail if the version does not match the latest known good version.
+  uint32 version = 80;
+
+  // The Credential Store type.
+  string type = 90;
+
+  // The attributes that are applicable for the specific Credential Store type.
+  google.protobuf.Struct attributes = 100 [(custom_options.v1.generate_sdk_option) = true];
+
+  // Output only. The available actions on this resource for this user.
+  repeated string authorized_actions = 300 [json_name = "authorized_actions"];
+
+  // Output only. The authorized actions for the scope's collections.
+  map<string, google.protobuf.ListValue> authorized_collection_actions = 310 [json_name = "authorized_collection_actions"];
+}
+
+// The attributes of a vault typed Credential Store.
+message VaultCredentialStoreAttributes {
+  // The complete url address of vault.
```

Comment:
What about the `token` and `client_certificate` attributes?

Reply:
I've added these fields.